### PR TITLE
[WIP] Adding documentation for addons_daily dataset

### PR DESCRIPTION
**Pull request by brianwright2**

### CODE REVIEW

**Comment by benmiroglio** [comment]
@brianwright2 Thanks for the work on these docs!

I haven't given a full read-through yet, but one thing I notice is the travis build is failing due to "misspellings" since it doesn't recognize certain words. To mitigate this, you should add any essential words to the `.spelling` file, which tells travis they are valid. 

The file is here for ref: https://github.com/mozilla/firefox-data-docs/blob/master/.spelling

**Comment by brianwright2** [comment]
@benmiroglio spelling changes fixed

**Comment by benmiroglio** [comment]
@brianwright2 You need to also add your article to the [`SUMMARY.md`](https://github.com/mozilla/firefox-data-docs/blob/master/src/SUMMARY.md) file. This should go under Derived Datasets.

Its probably worth following these local build instructions [here](https://github.com/mozilla/firefox-data-docs/blob/master/src/meta/contributing.md#building-the-documentation), so you can render a local version of the docs with your changes to verify things looks the way you want (this is also how I'll be reviewing them, which is how I noticed the missing index).

File: .spelling [hunk 1]
```diff
@@ -11,6 +11,7 @@ adb
 add-on
 addon
 addons
+addons_daily
```
Comment (mreid-moz) [inline .spelling]: drive-by: technical terms or code type things should be surrounded by backticks in the main text, which will make them pass the spell checker - `addons_daily`, `main_summary` and `v1` fall into this category.

Comment (brianwright2) [reply .spelling]: Thanks for the heads up!

File: src/datasets/other/addons_daily/reference.md [hunk 2]
```diff
@@ -3,17 +3,122 @@
 # `addons_daily` Derived Dataset
 Contributers: Sarah Melancon, Ben Miroglio, Brian Wright, Daniel Thorn
 
-This ETL code produces daily aggregates of Firefox extensions. It supports the broader "Extention Data for Developers" Project.
+This ETL code produces daily aggregates of Firefox extensions. It supports the broader "Extension Data for Developers" Project.
```
Comment (benmiroglio) [inline src/datasets/other/addons_daily/reference.md]: This is good for the repo README, but the context is a bit lost in the DTMO docs. Let's remove this sentence and focus on the dataset's details for DTMO.

File: src/datasets/other/addons_daily/reference.md [hunk 3]
```diff
@@ -4,16 +4,16 @@ This ETL code produces daily aggregates of Firefox extensions. It supports the b
...
 
-The addons_daily dataset serves as the central hub for all Firefox extension related questions. This includes questions regarding browser performance, user engagement, click through rates, etc. Each row in the table represents a unique addon, and each column is a unique metric.
+The `addons_daily` dataset serves as the central hub for all Firefox extension related questions. This includes questions regarding browser performance, user engagement, click through rates, etc. Each row in the table represents a unique addon, and each column is a unique metric.
 
 ### Contents
-Prior to construction of this dataset, extension related data lived in several different sources. Addons_daily has combined metrics aggregated from several sources, including raw pings, telemetry data, and google analytics data.
+Prior to construction of this dataset, extension related data lived in several different sources. `Addons_daily` has combined metrics aggregated from several sources, including raw pings, telemetry data, and google analytics data.
 
 ### Accessing the Data
 The data is stored as a parquet table in S3 [here](s3://net-mozaws-prod-us-west-2-pipeline-analysis/bmiroglio/addons_daily_test/)
```
Comment (benmiroglio) [inline src/datasets/other/addons_daily/reference.md]: Its better to just display the path rather than link it, mainly for copy+paste reasons. Also the path for the productionized dataset is `s3://telemetry-parquet/addons_daily/v1/`.

File: src/datasets/other/addons_daily/reference.md [hunk 4]
```diff
@@ -3,17 +3,122 @@
...
+## Data Reference
+
+### Example Queries
+
+
+#### Query 1
+
+
+```
+SELECT addon_id,
```
Comment (benmiroglio) [inline src/datasets/other/addons_daily/reference.md]: For the example queries, write a sentence explaining the interpretation of the query. Since the data is in re:dash now, you should link to a valid re:dash query as well.

**Comment by benmiroglio** [comment]
Thanks @brianwright2. One last thing:

The second query is a bit misleading, since this involves system and experiment add-ons as well. I think a common use case will be to look for dau over time for a *specific* add-on, so I think we can adjust that query to something like:

```
SELECT DATE_PARSE(submission_date_s3, '%Y%m%d') as "Date",
       dau as "DAU"
FROM addons_daily
WHERE
    addon_id = 'uBlock0@raymondhill.net'
```

**Comment by benmiroglio** [comment]
Also, let's mention in that this dataset is a 1% of Release Firefox

**Comment by benmiroglio** [comment]
LGTM, thanks again for writing these docs!
